### ISSUE TITLE
Deletion of a list entry from operational datastore

### ISSUE THREAD

**Issue by meghna-doshi** [sysrepo/sysrepo#2690]
Hi,
We have one query, to delete a list entry and whole subtree under it, we have used delete_item API as below

```
delete_item (xpath)
ex xpath:
/o-ran-aggregation-base:aggregated-o-ru/aggregation[ru-instance='rpi1'] 

```
is the above use of delete_item fine or do we have to give xpath upto the key element i.e /o-ran-aggregation-base:aggregated-o-ru/aggregation[ru-instance='rpi1']/ru-instance

sysrepo version used:
sysrepoctl - sysrepo YANG schema manipulation tool, compiled with libsysrepo v1.4.147 (SO v5.6.63)

**Comment by michalvasko** [comment]
No, what you wrote is fine and will delete the list instance with all its descendants.

**Comment by meghna-doshi** [comment]
Hi Michal,
Ok thanks. What would happen if we are using xpath as follows in delete_item, do you see any issue with this?

/o-ran-aggregation-base:aggregated-o-ru/aggregation[ru-instance='rpi1']/ru-instance

**Comment by michalvasko** [comment]
I actually was not sure what happens in such a case but I learnt that such edit are silently ignored. That did not seem right so I have added explicit checks that will return an error instead. So, please do not edit list keys directly, only list instances themselves.
